packages/react-radio/src/components/RadioGroup/RadioGroup.test.tsx

```diff
@@ -10,14 +10,207 @@ describe('RadioGroup', () => {
     displayName: 'RadioGroup',
   });
 
-  // TODO add more tests here, and create visual regression tests in /apps/vr-tests
+  it('renders a default state', () => {
+    render(<RadioGroup />);
+    expect(screen.getByRole('radiogroup')).toBeTruthy();
+  });
 
-  beforeEach(() => {
-    resetIdsForTests();
+  it('applies name to every radio item', () => {
+    render(
+      <RadioGroup name="test-name">
+        <Radio />
+        <Radio />
+        <Radio />
+      </RadioGroup>,
+    );
+    const items = screen.getAllByRole<HTMLInputElement>('radio');
+    expect(items[0].name).toBe('test-name');
+    expect(items[1].name).toBe('test-name');
+    expect(items[2].name).toBe('test-name');
   });
 
-  it('renders a default state', () => {
-    const result = render(<RadioGroup />);
-    expect(result.container).toMatchSnapshot();
+  it('applies the same generated name to every radio item if a name is not given', () => {
+    render(
+      <RadioGroup>
+        <Radio />
+        <Radio />
+        <Radio />
+      </RadioGroup>,
+    );
+    const items = screen.getAllByRole<HTMLInputElement>('radio');
+    expect(items[0].name).toBeTruthy();
+    expect(items[1].name).toBe(items[0].name);
+    expect(items[2].name).toBe(items[0].name);
+  });
+
+  it('respects radio item name', () => {
```

Comment:
This test highlights a potential bug. Because of the way `onChange` is implemented in `RadioGroup`, defining `name` on each child `Radio` prevents the onChange prop handler from being called as the `RadioGroup` name will be different from the `Radio` name.

Here's the `RadioGroup` onChange handler:

```tsx
onChange: useEventCallback(ev => {
        if (onChange && ev.target instanceof HTMLInputElement && ev.target.name === name) {
          onChange(ev, { value: ev.target.value });
        }
      }),
```

Reply:
Great catch. The reason for the name check in onChange was to avoid calling onChange if there was another input like `<Input>` or `<Dropdown>` on a radio button (e.g. an "Other" radio button with a text field). 

Perhaps a better way to do that check is to check for `type === 'radio'` instead of checking the name. Theoretically someone could add another RadioGroup inside the RadioGroup and break it, but that is pure madness (aka incorrect usage).

Reply:
I updated the implementation of `onChange`, and added a test for the scenario you described.